src/tests/test_crystals.cpp

```diff
@@ -0,0 +1,483 @@
+                            res.test_is_eq<size_t>("requested size", polys.size(), 4);
+
+                            for(const auto& poly : polys) {
+                               res.confirm("poly embedded in vector does not own memory", !poly.owns_storage());
+                            }
+
+                            Kyberish_PolyVec<Domain::NTT> polys_ntt(4);
+                            res.test_is_eq<size_t>("requested size (NTT)", polys.size(), 4);
+
+                            for(const auto& poly : polys_ntt) {
+                               res.confirm("poly (NTT) embedded in vector does not own memory", !poly.owns_storage());
+                            }
+                         }),
+
+      Botan_Tests::CHECK("cloned polynomials always manage their storge",
+                         [](Test::Result& res) {
+                            Kyberish_Poly<Domain::Normal> p;
+                            auto p2 = p.clone();
+                            res.confirm("cloned poly owns memory", p2.owns_storage());
+
+                            Kyberish_PolyVec<Domain::Normal> pv(3);
+                            for(auto& poly : pv) {
+                               res.require("poly in vector does not own memory", !poly.owns_storage());
+                               auto pv2 = poly.clone();
+                               res.confirm("cloned poly in vector owns memory", pv2.owns_storage());
+                            }
+
+                            auto pv2 = pv.clone();
+                            for(const auto& poly : pv2) {
+                               res.confirm("cloned vector polynomial don't own memory", !poly.owns_storage());
+                            }
+
+                            Kyberish_Poly<Domain::NTT> p_ntt;
+                            auto p2_ntt = p_ntt.clone();
+                            res.confirm("cloned poly (NTT) owns memory", p2_ntt.owns_storage());
+
+                            Kyberish_PolyVec<Domain::NTT> pv_ntt(3);
+                            for(auto& poly : pv_ntt) {
+                               res.require("poly (NTT) in vector does not own memory", !poly.owns_storage());
+                               auto pv2_ntt = poly.clone();
+                               res.confirm("cloned poly (NTT) in vector owns memory", pv2_ntt.owns_storage());
+                            }
+
+                            auto pv2_ntt = pv_ntt.clone();
+                            for(const auto& poly : pv2_ntt) {
+                               res.confirm("cloned vector polynomial (NTT) don't own memory", !poly.owns_storage());
+                            }
+                         }),
+
+      Botan_Tests::CHECK("hamming weight of polynomials",
+                         [](Test::Result& res) {
+                            Kyberish_Poly<Domain::Normal> p;
+                            res.test_is_eq<size_t>("hamming weight of 0", p.hamming_weight(), 0);
```

Comment:
already tested below

Reply:
One is testing poly, the other polyvec. The more the merrier I would say.